### PR TITLE
Get following siblings for certain nodes

### PR DESCRIPTION
**Pull request by marmoure**
Create a new function `followingSiblings` to enable getting next possible siblings.

### CODE REVIEW

File: packages/lwdita-ast/src/nodes/base.ts [hunk 1]
```diff
@@ -299,6 +307,29 @@ export abstract class AbstractBaseNode implements BaseNode {
...
+    // check where the child is in the list of allowed childTypes
+    for(const childType of childTypes) {
+      if(acceptsNodeName(child, childType, nodeGroups)) {
+        targetChildType = childType;
+        break;
+      }
+    }
+
+    // if the child is not in the list of allowed children, throw an error
+    if(!targetChildType) throw new Error(`Child ${child} not found in allowed children`);
```
Comment (adamretter) [inline packages/lwdita-ast/src/nodes/base.ts:324]: I think we should avoid the use of exceptions, and instead change the return type of the function so the return value is optional.

File: packages/lwdita-ast/src/nodes/base.ts [hunk 2]
```diff
@@ -299,6 +307,29 @@ export abstract class AbstractBaseNode implements BaseNode {
...
+
+    // get the index of the child in the list of allowed children and return the following siblings
+    const index = childTypes.indexOf(targetChildType);
+    // if the childType is a group include it in the list  
+    if((targetChildType as ChildType).isGroup || !(targetChildType as ChildType).single) return childTypes.slice(index);
```
Comment (adamretter) [inline packages/lwdita-ast/src/nodes/base.ts:329]: Split this into multiple lines with braces please.

File: packages/lwdita-ast/src/nodes/base.ts (hunk 1 above)
Comment (adamretter) [inline packages/lwdita-ast/src/nodes/base.ts:324]: Split this into multiple lines with braces please.

File: packages/lwdita-ast/test/audio.spec.ts [hunk 3]
```diff
@@ -70,4 +70,50 @@ describe('Class AudioNode', () => {
     expect(audio.muted).to.equal("false");
     expect(audio.tabindex).to.equal("1");
   });
+});
+
+describe('followingSiblings', () => {
+  it('get expected children for audio after desc', () => {
```
Comment (adamretter) [inline packages/lwdita-ast/test/audio.spec.ts:76]: 'get expected children for audio after desc' -> 'get the followingSiblings of `desc` in the `audio` element'.

Rename other tests in this fashion too please.

File: packages/lwdita-ast/src/nodes/base.ts [hunk 4]
```diff
@@ -194,7 +202,7 @@ export abstract class AbstractBaseNode implements BaseNode {
       return this.canAdd(childNodeName);
   }
 
-  private canAdd(childNodeName: string): boolean {
+  canAdd(childNodeName: string): boolean {
```
Comment (adamretter) [inline packages/lwdita-ast/src/nodes/base.ts:205]: Why have we changed the visibility of this method? It looks like nothing else in the PR uses this?

File: packages/lwdita-ast/src/nodes/base.ts (hunk 1 above)
Comment (adamretter) [reply packages/lwdita-ast/src/nodes/base.ts:324]: Also we should add a test or two for when it returns an empty value.

**Review by adamretter — CHANGES_REQUESTED** [review]

File: packages/lwdita-ast/src/nodes/base.ts (hunk 4 above)
Comment (marmoure) [reply packages/lwdita-ast/src/nodes/base.ts:205]: This was a mistake, I will revert it.

**Review by adamretter — CHANGES_REQUESTED** [review]
Small change to the function so it is single responsibility.

File: packages/lwdita-ast/src/nodes/base.ts [hunk 5]
```diff
@@ -299,6 +307,35 @@ export abstract class AbstractBaseNode implements BaseNode {
     // by YB, signed-off by AR.
     return this.canAdd("text") || this.canAdd("alt");
   }
+
+  followingSiblings(child?: string): ChildTypes[] | undefined {
```
[the remark this answers is not in the document]
Comment (marmoure) [reply packages/lwdita-ast/src/nodes/base.ts]: Fixed in [b38ec70](https://github.com/evolvedbinary/lwdita/pull/370/commits/b38ec700d415ab0b2acf6949d554d5f6c7411edb)
Thanks for the design pattern tip.

**Review by adamretter — APPROVED** [review]

**Review by adamretter — APPROVED** [review]
